src/zombies.test.ts

```diff
@@ -1,30 +1,121 @@
-import { ok } from "node:assert/strict";
+import { ok, equal } from "node:assert/strict";
 import { test } from "node:test";
 
+const numIsValidInteger = (num: unknown): num is number => {
```

Comment:
Thanks for good comments Marcus! 

I don’t quite follow though where I “lose track of the type validation”? Let me please explain how I've been thinking. 

My first type guard **numIsValidInteger** does an initial check in the function that the room is created with a valid integer: 0, 1, 2, 3... If not, an error will be thrown (for -1, 1.23 etc.). In my code I’ve just visualized this by returning an error message. Since my function can return either an “error” or a _ZombiereturnObj_ I need to cast the return value to _ZombiereturnObj_ in my tests (which can be done since I create the room with valid integers). 

My second type guard **isValidString** checks that the zomibe _added_ to the room is a string and that it is not the empty string. Doing this check and there will always be a zombie of correct format added. Hence the rest of the app doesn’t explicitly need to know about this type validation. I would be happy to hear your comments on this since I may have overlooked something.